### PR TITLE
Stories Carousel design changes #545

### LINKED ISSUE #545: CR: Stories Carousel design changes
From vg-volvotrucks-us-rd created by [cogniSyb](https://github.com/cogniSyb): Netcentric/vg-volvotrucks-us-rd#78  

Discussed with client: remove the author from "Stories from the road" carousel due to having no proper content available.

A new, [more consistent design has been created](https://www.figma.com/file/Id3nd9kbfh8TpS1SfWhrW2/Volvo-Component-Library?type=design&node-id=424-8853&mode=design&t=6tco0LYQlMAQBctt-4) for the navigation of the carousel. This is shared with #70 

### Requirements

- The title should be changed, everything behind the pipe character should be stripped 
- Buttons with previous/next arrows are used to navigate
  - When the start or the end of the carousel is reached, the respective button will be shown as inactive
  - Buttons are only present on tablet and desktop
  - On tablet, the navigation buttons are placed on the sides of the text
  - On desktop, the navigation buttons are placed near each other
- Text will be truncated if too long
  - If the title text exceeds 1 line, text will truncate
  - If the body copy text exceeds 2 lines, text will truncate
- Retrieve a maximum of articles, [as mentioned](https://github.com/hlxsites/vg-volvotrucks-us/issues/600#issuecomment-2051740862) in #600
- This block can be displayed on article pages and home pages 

Desktop: 
image
Tablet: 
image
Mobile: 
image

### PR DESCRIPTION
**Pull request by amglebihan**
Fix #545

Test URLs:
- Before: https://main--vg-volvotrucks-us--hlxsites.aem.page/block-library/blocks/v2-stories-carousel
- After: https://545-stories-carousel-changes--vg-volvotrucks-us--hlxsites.aem.page/block-library/blocks/v2-stories-carousel

### CODE REVIEW

File: blocks/v2-stories-carousel/v2-stories-carousel.css [hunk 1]
```diff
@@ -241,24 +255,26 @@ ul.v2-stories-carousel-meta {
...
-  .v2-stories-carousel::before {
-    content: '';
-    display: block;
-    height: 100%;
-    width: 30px;
-    position: absolute;
-    top: 0;
-    pointer-events: none;
+  .v2-stories-carousel-text {
+    max-width: 456px;
```
Comment (jonatan-lledo-netcentric) [inline blocks/v2-stories-carousel/v2-stories-carousel.css:259]: Where did this magic number come from?

**Review by alexiscoelho — APPROVED** [review]

File: blocks/v2-stories-carousel/v2-stories-carousel.js [hunk 2]
```diff
@@ -113,27 +129,18 @@ const buildStoryCard = (entry) => {
...
-  </svg></span>`;
   const svgArrowRight = `<span class="icon icon-arrow-right"><svg width="24" height="24" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Comment (jonatan-lledo-netcentric) [inline blocks/v2-stories-carousel/v2-stories-carousel.js:144]: this svg can be added as svg file and decorated later using decorate icons?

File: blocks/v2-stories-carousel/v2-stories-carousel.css (hunk 1 above)
Comment (amglebihan) [reply blocks/v2-stories-carousel/v2-stories-carousel.css:259]: From Figma, the text block has this width.

Comment (jonatan-lledo-netcentric) [reply blocks/v2-stories-carousel/v2-stories-carousel.css:259]: let's keep it like this then 👍🏻

File: blocks/v2-stories-carousel/v2-stories-carousel.js [hunk 3]
```diff
@@ -46,59 +47,74 @@ const getScrollOffset = (carousel) => {
...
     }
   }
 
-  setCarouselPosition(carousel, index);
+  setCarouselPosition(carousel, index, arrowLeftButton, arrowRightButton);
+};
+
+const createArrowButton = (direction) => {
+  const button = createElement('button', { classes: [`${blockName}-arrowbutton`] });
+  button.setAttribute('aria-label', direction === 'left' ? 'Previous' : 'Next');
```
Comment (jonatan-lledo-netcentric) [inline blocks/v2-stories-carousel/v2-stories-carousel.js:91]: the createElement has another option that remove the need of use later the setAttribut like this:
```
const button = createElement('button', {
  classes: [`${blockName}-arrowbutton`] },
  props: { 'aria-label': direction === 'left' ? 'Previous' : 'Next'
);
```

File: blocks/v2-stories-carousel/v2-stories-carousel.js (hunk 2 above)
Comment (amglebihan) [reply blocks/v2-stories-carousel/v2-stories-carousel.js]: Changed, we already had the SVG file, and I changed how the icon is added.

File: blocks/v2-stories-carousel/v2-stories-carousel.js (hunk 3 above)
Comment (amglebihan) [reply blocks/v2-stories-carousel/v2-stories-carousel.js]: Changed. I removed createArrowButton.

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Review by alexiscoelho — APPROVED** [review]

**Review by santi-homps — APPROVED** [review]

**Review by jonatan-lledo-netcentric — APPROVED** [review]

**Review by santi-homps — APPROVED** [review]

**Pull request merged**